src/lsp/swift-mesonlsp.ts

```diff
@@ -0,0 +1,64 @@
+import * as os from "os";
+
+import {
+  LanguageClient,
+  LanguageClientOptions,
+  ServerOptions,
+  Executable,
+  TransportKind
+} from "vscode-languageclient/node";
+import {
+  ExtensionContext,
+} from "vscode"
+import { LanguageServerClient } from "../lsp";
+
+export class SwiftMesonLspLanguageClient extends LanguageServerClient {
+  ls: LanguageClient | null = null
+
+  constructor(ctx: ExtensionContext) {
+    super("Swift-MesonLSP", ctx);
+  }
+
+  get downloadInfo(): [url: string, hash: string] | null {
+    const platform = os.platform();
+    if (platform != "win32" && platform != "darwin")
+      return null;
+    const arch = os.arch();
+    if (arch != "x64")
+      return null;
+    if (platform == "win32")
+      return ["https://github.com/JCWasmx86/Swift-MesonLSP/releases/download/v2.0/Swift-MesonLSP-win64.zip",
+              "fc0d930eb67525309291907c07df8655920cdc651f17f2320d09cc5271a23876"]
+    return ["https://github.com/JCWasmx86/Swift-MesonLSP/releases/download/v2.0/Swift-MesonLSP-macos12.zip",
+            "c10999081cef8b56ea164d9a170470c268045254829e48bf4ac137a07a8a874c"]
+  }
+
+  supportsSystem(os: string, arch: string): boolean {
+    if (arch != "x64")
+      return false;
+    return os == "win32" || os == "darwin";
+  }
+
+  startLanguageServer(serverModule: string): void {
```

Comment:
I think this function can be a private function to the abstract parent class. The `clientOptions` should probably be a private member of that base class. I don't see it changing for other potential clients. Do you?

I think `runExe` and `debugExe` could probably be abstract getters on the base class? Then `serverOptions` could be made using the abstract getters.

Overall, I think this design is coming together quite nicely.

Reply:
debugExe/runExe depend on `serverModule`, so best I can do is a method that e.g. returns an array of executables. Or we could mandate, that all implementations have to accept `--lsp`?